application/forms/Authentication/LoginForm.php

```diff
@@ -98,6 +111,21 @@ public function onSuccess()
         $authenticated = $authChain->authenticate($user, $password);
         if ($authenticated) {
             $auth->setAuthenticated($user);
+            if ($this->getElement('rememberme')->isChecked()) {
+                $rsa = new RSA();
+                $rsa->loadKey(...RSA::keygen());
+                $data = $rsa->encryptToBase64($user->getUsername(), $password);
+                $data[] = base64_encode($rsa->getPublicKey());
+
+                $values['username'] = $user->getUsername();
```

Comment:
One last change to this block here and it's good:

You don't have to cache the values but may passt it directly to insert, i.e.

```php
$this->getDb()->insert('rememberme', [
    ...
]);

Reply:
And, you have to pass the timestamp for `ctime` and `mtime`.